lib/web_ui/test/engine/platform_views/content_manager_test.dart

```diff
@@ -0,0 +1,170 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:html' as html;
+
+import 'package:ui/src/engine.dart';
+
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+
+import '../../matchers.dart';
+
+void main() {
+  internalBootstrapBrowserTest(() => testMain);
+}
+
+void testMain() {
+  group('PlatformViewManager', () {
+    final String viewType = 'forTest';
+    final int viewId = 6;
+
+    late PlatformViewManager contentManager;
+
+    setUp(() {
+      contentManager = PlatformViewManager();
+    });
+
+    group('knowsViewType', () {
+      test('recognizes viewTypes after registering them', () async {
+        expect(contentManager.knowsViewType(viewType), isFalse);
+
+        contentManager.registerFactory(viewType, (int id) => html.DivElement());
+
+        expect(contentManager.knowsViewType(viewType), isTrue);
+      });
+    });
+
+    group('knowsViewId', () {
+      test('recognizes viewIds after *rendering* them', () async {
+        expect(contentManager.knowsViewId(viewId), isFalse);
+
+        contentManager.registerFactory(viewType, (int id) => html.DivElement());
```

Comment:
test that `knowsViewId` is false after registering but before `renderContent`

Reply:
Added an expectation right below the registerFactory to check that viewsId is not known, good catch.